src/classes/Pantry.js

```diff
@@ -0,0 +1,50 @@
+class Pantry {
+  constructor(currentUser) {
+    this.pantry = currentUser.pantry;
+  };
+
+  checkCookability(recipe, currentUser) {
+    let ingredientStatus = recipe.checkPantryForIngredientAmounts(currentUser);
+
+    //notEnoughInPantry is an array being filled with ingredients of which we don't have enough
+
+    let notEnoughInPantry = ingredientStatus.filter(ingredient => ingredient.difference < 0);
+
+    //if there is ANYTHING in notEnoughInPantry, then the recipe can't be cooked, and false is returned
+
+    return !notEnoughInPantry.length ? true : false;
+  };
+
+  removeFromPantry(recipe, currentUser) {
+    let thisPantry = recipe.ingredientInfo.filter(ingred => {
+      this.pantry.forEach(ingredient => {
+        if (ingredient.ingredient === ingred.id) {
+          ingredient.amount -= ingred.quantity;
+        };
+      });
+      return this.pantry;
+    });
+    console.log('this pantry:', this.pantry);
+  };
+
+  addToPantry(inputValue, ingredientId, currentUser) {
+    this.pantry.forEach(ingredient => {
+      if (ingredient.ingredient === ingredientId) {
+        ingredient.amount += inputValue;
+      }
+    });
+
+    let pantryIngredientIds = this.pantry.reduce((acc, ingredient) => {
+      acc.push(ingredient.ingredient);
+      return acc;
+    }, []);
+
+    if (!pantryIngredientIds.includes(ingredientId)) {
```

Comment:
I'm wondering if there's a way to include this logic inside the code block that starts on 31.